src/devices/Fsr408/Fsr408.cs

```diff
@@ -0,0 +1,154 @@
+
+        public Fsr408()
+        {
+            _controller = new GpioController();
+            _controller.OpenPin(_pinNumber);
+        }
+        public Fsr408(int pinNumber)
+        {
+            _pinNumber = pinNumber;
+            _controller = new GpioController();
+            _controller.OpenPin(_pinNumber);
+        }
+
+
+        public int ReadFromMcp3008()
+        {
+            if (AdcConverter != null)
+            {
+                return AdcConverter.Read(0);
+            }
+            else
+            {
+                throw new NotSupportedException("ADC converter not set");
+            }
+        }
+
+        public int ReadCapacitorChargingDuration()
+        {
+            
+            // set pin to low
+            _controller.SetPinMode(_pinNumber, PinMode.Output);
+            _controller.Write(_pinNumber, PinValue.Low);
+
+            // Prepare pin for input and ...
+            _controller.SetPinMode(_pinNumber, PinMode.Input);
+            Stopwatch timeElapsed = Stopwatch.StartNew();
+            while (_controller.Read(_pinNumber) == PinValue.Low)
+            { // count until read high
+                
+                if (timeElapsed.ElapsedMilliseconds == 30000)
+                {   // if count goes too high it means FSR resustance is highest which means no pressure, don't need to count more  
+                    break;
+                }
+            }
+            return (int)timeElapsed.ElapsedMilliseconds;
+        }
+
+        public int ReadVotlageUsingMcp3008()
+        {
+            if (AdcConverter != null)
+            {
+                int readValue = AdcConverter.Read(0);
+                return CalculateVoltage(readValue);
+            }
+            else
+            {
+                throw new NotSupportedException("ADC converter not set");
+            }
+        }
+
+        public int CalculateVoltage(int readValue)
+        {
+            // Mcp3008 analog voltage reading ranges from 0 to 1023 (10 bit) 
+            // mapping it to corresponding milli voltage 
+            // which ranging from 0mV to PowerSupplied mV 
+            return PowerSupplied * readValue / 1023;
+        }
+
+        public int ReadFsrResistanceUsingMcp3008()
+        {
+            if (AdcConverter != null)
+            {
+                int readValue = AdcConverter.Read(0);
+                return CalculateFsrResistance(CalculateVoltage(readValue));
+            }
+            else
+            {
+                throw new NotSupportedException("ADC converter not set");
+            }
+        }
+
+        public int CalculateFsrResistance(int fsrVoltage)
+        {
+            // Formula: FSR = ((Vcc - V) * R) / V
+            if (fsrVoltage > 0)
+            {
+                return (PowerSupplied - fsrVoltage) * Resistance / fsrVoltage;
+            }
+            return 0;
+        }
+
+        public int ReadPressureForceUsingMcp3008()
```

Comment:
All of these methods, I would not use Mcp3008 on the method name, I would just have a method called `ReadPreassureForce` and just have an enum inside the class defining the connection type, and then inside this implementation I would just do a switch depending on that connection type. That way, if somebody already has an app and simply want to change the way the device is hooked up, they would only need to change one line (the constructor) instead of changing all of their interactions with the API

Reply:
Agree, but as i mentioned above read values from ADC and capacitor are not matching or interchangeable. That is why i wondering if its better to separated these bindings